app.py

```
@@ -0,0 +1,190 @@
+    
+
+    def update_subscriber(self):
+     req = request.get_json()
+     list_name = req['list_name']
+     user_email = req['user_email']      
+     try:
+      list_id = self.get_list_id(list_name)
+     except Exception as e:
+      return self.not_found_error(e)
+     try:
+      user_id = self.get_user_id(list_id, user_email)
+     except Exception as e:
+      return self.not_found_error(e)
+
+     data = self.subscriber_data()
+   
+     try:
+       self.client.lists.members.update(list_id, user_id, data)
+       return self.end({})
+     except Exception as e:
+       return self.end({'success': False, 'error_code': e.args[0]['status'], 'error': e.args[0]['detail']})
+
+
+    #Prepares and returns subscriber's data as a dictionary
+    def subscriber_data(self):
+     req = request.get_json()
+     data = {}
+     merge_field_data = {}
+   
+     key_map = {
+       'status': 'status',
+       'first_name': 'FNAME', 
+       'last_name': 'LNAME',
+       'new_email': 'email_address',
+       'address': 'ADDRESS',
+       'phone': 'PHONE'
+     }
+   
+     merge_field_exclusive = ['first_name', 'last_name', 'address', 'phone']
+
+     for key, val in req.items():
+       if key != 'list_name' and key != 'user_email' and key not in merge_field_exclusive:
+         data.update({key_map[key]:val})
+       elif key != 'list_name' and key != 'user_email':
+         merge_field_data.update({key_map[key]:val})
+   
+     data.update({'merge_fields': merge_field_data})
+
+     return data
+
+
+    # Returns List ID from List name input 
+    def get_list_id(self, list_name):
+     found = False
+     list_name = list_name
+     for x in self.client.lists.all(get_all=True, fields="lists.name,lists.id")['lists']:
+       if(x['name'] == list_name):
+         return x['id']
+         found = True
+   
+     if not found:
+      raise Exception('Invalid list name.')
```

Better create your own exception type, s.t. you don't need to catch all exceptions above.